collect_app/src/main/java/org/odk/collect/android/logic/AuditEvent.java

```diff
@@ -61,20 +62,32 @@ public String getValue() {
 
     private final long start;
     private AuditEventType auditEventType;
-    private final String node;
+    @NonNull private final String node;
     private String latitude;
     private String longitude;
     private String accuracy;
+    @NonNull private String oldValue;
+    @NonNull private String newValue = "";
     private long end;
     private boolean endTimeSet;
+    private boolean isTrackingChangesEnabled;
+    private FormIndex formIndex;
 
     /*
      * Create a new event
      */
-    public AuditEvent(long start, AuditEventType auditEventType, String node) {
+    public AuditEvent(long start, AuditEventType auditEventType) {
+        this(start, auditEventType, false, null, null);
+    }
+
+    public AuditEvent(long start, AuditEventType auditEventType, boolean isTrackingChangesEnabled,
+                      FormIndex formIndex, String oldValue) {
         this.start = start;
         this.auditEventType = auditEventType;
-        this.node = node;
+        this.isTrackingChangesEnabled = isTrackingChangesEnabled;
+        this.formIndex = formIndex;
+        this.oldValue = oldValue == null ? "" : oldValue;
+        this.node = formIndex == null || formIndex.getReference() == null ? "" : formIndex.getReference().toString();
```

Comment:
You don't need `node` as a field, right? You could move this to to a local variable in `toString` and use `formIndex` in other places where forms need to be identified.

I would expect that on print there's still an extra multiplicity identifier (`[1]`) at the end of the reference so that the code at https://github.com/opendatakit/collect/pull/3072/files#diff-6b2bb0a16e468e6adfa987b33be6e4fcL64 would still be needed. Is that not the case? Again, I'd put that in the `toString`.

Reply:
Done.